packages/contracts-bedrock/scripts/DeployOPChain.s.sol

```diff
@@ -274,16 +274,17 @@ contract DeployOPChainOutput is BaseDeployIO {
         return _delayedWETHPermissionlessGameProxy;
     }
 
-    // -------- Assertions on chain architecture --------
+    // -------- Deployment Assertions --------
 
     function assertValidDeploy(DeployOPChainInput _doi) internal view {
-        assertValidSystemConfig(_doi);
+        assertValidDelayedWETHs(_doi);
+        assertValidDisputeGameFactory(_doi);
         assertValidL1CrossDomainMessenger(_doi);
+        assertValidL1ERC721Bridge(_doi);
         assertValidL1StandardBridge(_doi);
         assertValidOptimismMintableERC20Factory(_doi);
         assertValidOptimismPortal(_doi);
-        assertValidDisputeGameFactory(_doi);
-        assertValidDelayedWETHs(_doi);
+        assertValidSystemConfig(_doi);
```

Comment:
I sorted these alphabetically to make it easier to diff the list of checks here with the list of checks in `DeployImplementations.s.sol`. This made me notice that `assertValidL1ERC721Bridge` was missing here

Reply:
yikes.

That does get at the fact that these checks are hard to review and maintain. I can't think of a great way to improve on the situation though. Even if we had working code coverage, these aren't technically 'tests' so it's unclear if that would help us. 

Would it make sense to create a ticket for future work to use `vm.StateAccesses` to ensure that all writes are eventually read during assertions?

Reply:
Yea good call on tracking improvements here. I added this to https://github.com/ethereum-optimism/optimism/issues/11814